src/init/features/turtles/repo.ts

```diff
@@ -0,0 +1,173 @@
+import { cloudbuildOrigin } from "../../../api";
+import { FirebaseError } from "../../../error";
+import * as gcb from "../../../gcp/cloudbuild";
+import { logger } from "../../../logger";
+import * as poller from "../../../operation-poller";
+import * as utils from "../../../utils";
+import { promptOnce } from "../../../prompt";
+
+const gcbPollerOptions: Omit<poller.OperationPollerOptions, "operationResourceName"> = {
+  apiOrigin: cloudbuildOrigin,
+  apiVersion: "v2",
+  masterTimeout: 25 * 60 * 1_000,
+  maxBackoff: 10_000,
+};
+
+function extractRepoSlugFromURI(remoteUri: string): string | undefined {
```

Comment:
nit* not asking for test here, but I'd be happy with a example use case in the comments, e.g.

```
extractRepoSlugFromURL("AAAAAA")
=> "BBBBBBBBBBBBBB"
```